hazelcast-client/src/main/java/com/hazelcast/client/impl/statistics/Statistics.java

```diff
@@ -0,0 +1,246 @@
+    public final static int FEATURE_SUPPORTED_SINCE_VERSION = BuildInfo.calculateVersion(FEATURE_SUPPORTED_SINCE_VERSION_STRING);
+
+    private final boolean enabled;
+    private final HazelcastProperties properties;
+    private final ILogger logger = Logger.getLogger(this.getClass());
+    private final HazelcastClientInstanceImpl client;
+
+    private final boolean enterprise;
+
+    private final MetricsRegistry metricsRegistry = new MetricsRegistryImpl(Logger.getLogger(this.getClass()),
+            ProbeLevel.MANDATORY);
+    private final Map<String, String> staticStats = new HashMap<String, String>(1);
+    private PeriodicStatistics periodicStats;
+
+    /**
+     * Use to enable the client statistics collection.
+     * <p/>
+     * The default is false.
+     */
+    public static final HazelcastProperty ENABLED = new HazelcastProperty(CONFIGURATION_PREFIX + ".enabled", false);
+
+    /**
+     * The period in seconds the statictics runs.
+     * <p/>
+     * The default is 60 seconds.
+     */
+    public static final HazelcastProperty PERIOD_SECONDS = new HazelcastProperty(CONFIGURATION_PREFIX + ".period.seconds", 60, SECONDS);
+
+    public Statistics(HazelcastClientInstanceImpl client) {
+        this.properties = client.getProperties();
+        this.enabled = properties.getBoolean(ENABLED);
+        this.client = client;
+        this.enterprise = BuildInfoProvider.getBuildInfo().isEnterprise();
+    }
+
+    /**
+     * Registers all client statistics and schedules peridic collection of stats
+     */
+    public void start() {
+        if (!enabled) {
+            return;
+        }
+
+        long periodSeconds = properties.getSeconds(PERIOD_SECONDS);
+        if (periodSeconds <= 0) {
+            long defaultValue = Long.parseLong(PERIOD_SECONDS.getDefaultValue());
+            logger.warning("Provided client statistics " + PERIOD_SECONDS.getName()
+                    + " can not be less than or equal to 0. You provided " + periodSeconds
+                    + " seconds as the configuration. Client will use the default value of " + defaultValue + " instead.");
+            periodSeconds = defaultValue;
+        }
+
+        registerStatistics();
+
+        schedulePeriodicStatisticsSendTask(periodSeconds);
+
+        logger.info("Client statistics is enabled with period " + periodSeconds + " seconds.");
+    }
+
+    private ClientConnection getOwnerConnection() {
+        Address ownerConnectionAddress = client.getClientClusterService().getOwnerConnectionAddress();
+        if (null == ownerConnectionAddress) {
+            return null;
+        }
+        ClientConnection connection = (ClientConnection) client.getConnectionManager().getConnection(ownerConnectionAddress);
+        if (null == connection) {
+            return null;
+        }
+
+        int serverVersion = connection.getConnectedServerVersion();
+        if (serverVersion < FEATURE_SUPPORTED_SINCE_VERSION) {
```

Comment:
yes, that is correct. Alternatives:
1. No logging: But then how can we know if this was the reason of not getting the statistics populated.
2. Do the log only once per connection: But then this would require unnecessary lookup table and such.

Do you have any other suggestion?

1.